proposal.rst

```diff
@@ -268,7 +267,29 @@ Project Timeline
 .. any university classes or exams, vacations, etc), make sure you include them
 .. in your project timeline.
 
-FILLME
+* Weeks of bonding: get familiar with ranking algorithms, search for suitable datasets, assessment of the complexity of integrating large databases, and performance tests.
+
+In each next position assumes that testing and writing cross-platform part is included.
+
+* 1 Week - integrate datasets 
+
+* 2 Weeks of exams (situation with exams is very unstable now in my University, so I timeline may change in this case)
+
+* 1 Week - solve possible problems with the size of datasets
+
+* 3 Weeks - implement ListMLE, ListNet and SVM evaluation api
```

Comment:
We're expecting quite a bit more detail than this, particularly for a 3 week segment. What do you mean by an "evaluation API" in this case? How will it relate to the previous work on xapian-evaluation mentioned in the letor stabilisation project page (linked from the evaluation suite project description)? How will it related to the `Scorer` classes already in the letor codebase? What would be different about the support for ListMLE, ListNet and SVM? Are you going to tackle the outstanding PRs from last year's letor project, or is this just about evaluation (in which case why are you focussing it on the letor rankers, rather than all weighting approaches)?

It's probably worth taking a step back and explaining in your proposal how and why someone would use this. You say in your motivation answer that it will be "useful for users [who] want to understand Xapian abilities", but ideally an evaluation system would also allow them to choose which ranking algorithm (including letor Ranker) and parameters works best for their case — and to allow the Xapian developers to keep track of which ones work best for which common real-world examples, so we can choose appropriate defaults that work well in general use.

I think it'd be good to sketch out how someone would use this project. Is it a tool that people run? (A bit like xapian-evaluation.) Is it a set of classes in Xapian? (A bit like Scorer.) Or perhaps it's a tool built on internal classes, so people can use the metrics in their own tools, or can evaluate performance using an easy-to-run one.

Also, what datasets are you going to evaluate over? It seems natural that part of this project would involve running the evaluation suite, for which you'll need a corpus and some test queries (ideally pre-labelled). A problem that vivekp ran into was that different datasets have different formats, so either you need to have a single format that is documented that others can be transformed into, or maybe you'll need a library of support for different formats.